### ISSUE TITLE
Ref returns/locals: Extract interface does not generate ref-returning methods

### ISSUE THREAD

**Issue by dpoeschl** [dotnet/roslyn#8438]
Extract interface on 

``` C#
  class Program
  {
      public ref int M() { return ref M(); }
  }
```

**Expected**: The generated interface should contain `ref int M();`
**Actual**: The generated interface contains `int M();`, causing an error

**Comment by DustinCampbell** [comment]
@jaredpar: Moving this one to the compiler to fix.

**Comment by VSadov** [comment]
the code simply misses 'ref', while annoying, user can workaround it by adding ref

can be fixed in 2.1

**Comment by CyrusNajmabadi** [comment]
This has been fixed with https://github.com/dotnet/roslyn/pull/22285
